src/api/flaskr/service/profile/profile_manage.py

```
@@ -56,6 +57,64 @@ def get_next_corlor_setting(parent_id: str):
 
 
+def _collect_used_variables(app: Flask, shifu_bid: str) -> set[str]:
+    """
+    Collect variable names referenced across the latest mdflow content
+    for all draft outline items under a shifu.
+    """
+    with app.app_context():
+        latest_ids_subquery = (
+            db.session.query(
+                DraftOutlineItem.outline_item_bid,
+                func.max(DraftOutlineItem.id).label("latest_id"),
```

Ruff (0.14.13)

Undefined name `func`

(F821)